clusters/clusters_api.go

```diff
@@ -444,6 +444,9 @@ func (cluster Cluster) Validate() error {
 	if profile == "singleNode" && strings.HasPrefix(master, "local") && resourceClass == "SingleNode" {
 		return nil
 	}
+	if cluster.InstancePoolID != "" && strings.HasPrefix(master, "local") {
```

Comment:
This would need to be added here as well: https://github.com/databricks/terraform-provider-databricks/pull/3376/files#diff-f7640d3fe05ecaaecb2e8d116cd5ade2a1e736780b93341dc2ee6ef67b5a13b8R64.